validator/src/journal/chain.rs

```diff
@@ -852,74 +851,8 @@ impl<TEP: ExecutionPlatform + Clone + 'static, PV: PermissionVerifier + Clone +
-        let settings_view: SettingsView = self
-            .state_view_factory
-            .create_view(&block.state_root_hash)
-            .map_err(|db_err| {
-                let err_str = format!(
-                    "Failed to get state view for block {:?}: {:?}",
-                    block, db_err
-                );
-                warn!("{}", err_str);
-                ChainControllerError::ConsensusError(err_str)
-            })?;
-        let name = settings_view
-            .get_setting_str("sawtooth.consensus.algorithm.name", None)
-            .map_err(|settings_err| {
-                let err_str = format!(
-                    "Error getting sawtooth.consensus.algorithm.name from settings view: {:?}",
-                    settings_err
-                );
-                warn!("{}", err_str);
-                ChainControllerError::ConsensusError(err_str)
-            })?
-            .unwrap_or_else(|| String::from(""));
-        let version = settings_view
-            .get_setting_str("sawtooth.consensus.algorithm.version", None)
-            .map_err(|settings_err| {
-                let err_str = format!(
-                    "Error getting sawtooth.consensus.algorithm.version from settings view: {:?}",
-                    settings_err
-                );
-                warn!("{}", err_str);
-                ChainControllerError::ConsensusError(err_str)
-            })?
-            .unwrap_or_else(|| String::from(""));
-
-        let is_active_engine = self
-            .consensus_registry
-            .is_active_engine_name_version(&name, &version)
-            .map_err(|reg_err| {
-                let err_str = format!("Error verifying active engine: {:?}", reg_err);
-                warn!("{}", err_str);
-                ChainControllerError::ConsensusError(err_str)
-            })?;
-
-        if !is_active_engine {
-            match self.consensus_registry.get_active_engine_info() {
-                Ok(Some(engine_info)) => {
-                    self.consensus_notifier
-                        .notify_engine_deactivated(engine_info.connection_id);
-                }
-                Err(err) => {
-                    // Not a fatal error; just won't notify old engine that it's deactivated
-                    warn!("Error getting active engine info: {:?}", err);
-                }
-                _ => (),
-            }
-            self.consensus_registry
-                .activate_engine(&name, &version)
-                .map_err(|reg_err| {
-                    let err_str = format!("Error activating consensus engine: {:?}", reg_err);
-                    warn!("{}", err_str);
-                    ChainControllerError::ConsensusError(err_str)
-                })?;
-            self.consensus_notifier.notify_engine_activated(block);
-        } else {
-            self.consensus_notifier
-                .notify_block_commit(&block.header_signature);
-        }
+        self.consensus_notifier
```

Comment:
This commit notification should either be before the activation change, or mutually exclusive with an activation change. Otherwise the new engine will get a block commit notification right away, which it won't expect because it didn't tell the validator to commit this block.

Reply:
Added a commit for this.